mlserver/parallel/registry.py

```diff
@@ -56,6 +57,24 @@ def __init__(
 
         os.makedirs(self._settings.environments_dir, exist_ok=True)
 
+        # Register sigchld signal handler (saving original to restore it later)
+        self._original_sigchld_handler = signal.getsignal(signal.SIGCHLD)
+        signal.signal(
+            signal.SIGCHLD,
+            lambda *args: asyncio.create_task(self._handle_worker_stop(*args)),
+        )
+
+    async def _handle_worker_stop(self, signum, frame):
+        pid, exit_code = os.waitpid(-1, os.WNOHANG)
+        if pid == 0 or exit_code == 0:
+            # Worker terminated gracefully, nothing to do
+            return
+
+        await self._default_pool.on_worker_stop(pid, exit_code)
```

Comment:
question: whats the difference between default_pool and other pools?

Reply:
The default pool uses the same environment as the main MLServer process and it's always running. When a model doesn't specify a custom environment, it will just go to the default pool.